kits/arm/arm.py

```diff
@@ -0,0 +1,217 @@
+# -*- coding: utf-8 -*-
+import hebi
+from time import sleep, time, perf_counter
+import numpy as np
+
+import os
+import sys
+
+# ------------------------------------------------------------------------------
+# Add the root folder of the repository to the search path for modules
+root_path = os.path.abspath(os.path.join(os.path.dirname(__file__), '..', '..'))
+sys.path = [root_path] + sys.path
+# ------------------------------------------------------------------------------
+
+from util.math_utils import get_grav_comp_efforts
+from util.math_utils import gravity_from_quaternion
+
+class ArmParams(object):
+    
+    def __init__(self, family, moduleNames, hrdf):
+        self.family = family
+        self.hrdf = hrdf
+        self.moduleNames = moduleNames
+
+
+    
+class Arm():
+    
+    def __init__(self, params):
+        # Create arm from lookup
+        self.lookup = hebi.Lookup()
+        sleep(2)
+        print(self.lookup)
+        self.grp = self.lookup.get_group_from_names([params.family], params.moduleNames)
+        print(self.grp)
+        
+        # Create robot model
+        try:
+            self.model = hebi.robot_model.import_from_hrdf(params.hrdf)
+        except Exception as e:
+            print('Could not load HRDF: {0}'.format(e))
+            sys.exit()
+        
+        # Create comand and feedback
+        self.cmd = hebi.GroupCommand(self.grp.size)
+        self.fbk = hebi.GroupFeedback(self.grp.size)
+        self.prev_fbk = self.fbk
+        self.grp.get_next_feedback(reuse_fbk=self.fbk)
+        
+        # Zero our initial comand setting vars
+        self.pos_cmd = self.fbk.position
+        self.vel_cmd = np.zeros(self.grp.size)
+        self.accel_cmd = np.zeros(self.grp.size)
+        self.eff_cmd = np.zeros(self.grp.size)
+        
+        # Setup gravity vector for grav comp
+        self.gravity_vec = [0, 0, 1]
+        gravity_from_quaternion(self.fbk.orientation[0], output=self.gravity_vec)
+        self.gravity_vec = np.array(self.gravity_vec)
+        
+        # Zero times
+        self.time_now = time()
+        self.traj_start_time = self.time_now
+        self.duration = 0
+        self.t_traj = 0
+        
+        self.set_goal_first = True
+        self.at_goal = True
+        self.goal = "grav comp"
+        
+    
+    # Loads gains from given gains file    
+    def loadGains(self, gains_file):
```

Comment:
I think that this function should accept a gains object rather than a filename. There are plenty of cases where we would want to load some gains and modify them before setting (e.g. the impedance control example increases the effort kp by 2x).

That being said, this would be a good candidate for a helper function rather than part of the Arm API.